File: abc/296/d.cpp
#include <bits/stdc++.h>
#define rep(i, n) for (long long i = 0; i < (long long)(n); ++i)
using namespace std;
using ll = long long;


int main() {
  ll n, m;
  cin >> n >> m;

  constexpr ll inf = 1ll << 60;
  ll ans = inf;
  for (ll a = 1; a <= n; ++a) {
    ll b = (m+a-1) / a;
    if (b <= n) ans = min(ans, a*b);
    if (a > b) break;
  }

  if (ans == inf) ans = -1;
  cout << ans << '\n';
  return 0;
}

File: abc/296/d.py
import math


n, m = map(int, input().split())
inf = 10**24 + 1
ans = inf
limit = int(math.sqrt(m))

for a in range(1, limit+1):
    b = (m+a-1) // a
    
    if a < 1 or a > n: continue
    if b < 1 or b > n: continue
    if a*b < m: continue

    ans = min(ans, a*b)

if ans == inf:
    ans = -1
    
print(ans)